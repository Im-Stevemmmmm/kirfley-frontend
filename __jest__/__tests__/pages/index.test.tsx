import { render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Homepage from '../../../pages/index';

const getChildren = () => {
  return ({ children }) => {
    return children;
  };
};

jest.mock('next/link', () => getChildren());
jest.mock('next/head', () => getChildren());

it('renders correctly', () => {
  const container = document.createElement('div');
  ReactDOM.render(<Homepage />, container);

  expect(container).toMatchSnapshot();
});

describe('page content', () => {
  beforeEach(() => render(<Homepage />));

  it('displays a greeting message', () => {
    expect(screen.getByText('Fern', { selector: 'h1' })).toBeInTheDocument;
    expect(
      screen.getByText(
        'A free platform where you can pretty much say anything.'
      )
    ).toBeInTheDocument;
  });

  it('renders the sign up and log in buttons', () => {
    expect(screen.getByRole('button', { name: 'Sign Up' })).toBeInTheDocument;
    expect(screen.getByRole('button', { name: 'Log In' })).toBeInTheDocument;
  });
});
