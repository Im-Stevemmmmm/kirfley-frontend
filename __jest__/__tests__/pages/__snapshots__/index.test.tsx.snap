// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    id="root"
  >
    <title>
      Fern
    </title>
    <h1>
      Fern
    </h1>
    <h2
      class="wrappedText"
    >
      A free platform where you can pretty much say anything.
    </h2>
    <div
      id="authButtons"
    >
      <button>
        Sign Up
      </button>
      <button>
        Log In
      </button>
    </div>
  </div>
</div>
`;
